pandas/core/frame.py

```diff
@@ -8385,14 +8385,18 @@ def groupby(
         axis: Axis = 0,
         level: IndexLabel | None = None,
         as_index: bool = True,
-        sort: bool = True,
+        sort: bool | lib.NoDefault = no_default,
```

Comment:
As a user, I would interpret `NoDefault` to mean "Pandas may, or may not, sort the result, and I should not be relying on one behaviour or the other". Is that the right interpretation?

Reply:
pandas will sort the result, it's other DataFrame libraries that might not